### PR TITLE
DOCS-3994 Adds SwiftUI Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Adds single sourcing logic and generates a preview for a SwifUI doc page.

### Motivation

DOCS-3994

### Preview 

https://docs-staging.datadoghq.com/alai97/browser-rum-swift-update/real_user_monitoring/ios/swiftui/

### Additional Notes

Remember to update the staging branch back to `main` and update the `pull_config.yaml` file before merging!

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: config/_default/menus/menus.en.yaml [hunk 1]
```diff
@@ -2869,31 +2869,36 @@ main:
     parent: rum
     identifier: rum_ios
     weight: 3
+  - name: Swift
```
Comment (maxep) [inline config/_default/menus/menus.en.yaml:2872]:
```suggestion
  - name: SwiftUI
```

File: config/_default/menus/menus.en.yaml [hunk 2]
```diff
@@ -2869,31 +2869,36 @@ main:
...
+    url: real_user_monitoring/ios/swift/
```
Comment (maxep) [inline config/_default/menus/menus.en.yaml:2873]:
```suggestion
    url: real_user_monitoring/ios/swiftui/
```

File: config/_default/menus/menus.en.yaml [hunk 3]
```diff
@@ -2869,31 +2869,36 @@ main:
...
+    parent: rum_ios
+    identifier: rum_ios_swift
```
Comment (maxep) [inline config/_default/menus/menus.en.yaml:2875]:
```suggestion
    identifier: rum_ios_swiftui
```

**Review by maxep — CHANGES_REQUESTED** [review]
Thanks @alai97 🙏 

It will be best to rename `Swift` -> `SwiftUI`

**Review by hestonhoffman — APPROVED** [review]
Fixed the build issue in the other PR. Looks good

File: config/_default/menus/menus.en.yaml [hunk 4]
```diff
@@ -2874,31 +2874,36 @@ main:
     parent: rum
     identifier: rum_ios
     weight: 3
+  - name: SwiftUI
+    url: real_user_monitoring/ios/swiftui/
+    parent: rum_ios
+    identifier: rum_ios_swiftui
+    weight: 301
```
Comment (maxep) [inline config/_default/menus/menus.en.yaml:2881]: I think we can move this down the `Advanced Configuration` section, makes sense to have this close to `Mobile Vitals` and `Web View Tracking`

**Review by maxep — APPROVED** [review]
Looks great 🚀

**Pull request merged**